libcloudforensics/providers/azure/internal/common.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common utilities."""
+
+import binascii
+import logging
+import os
+import re
+
+from typing import Any, List, Dict, Optional, TYPE_CHECKING
+from azure.common.credentials import ServicePrincipalCredentials  # pylint: disable=import-error
+
+if TYPE_CHECKING:
+  # TYPE_CHECKING is always False at runtime, therefore it is safe to ignore
+  # the following cyclic import, as it it only used for type hints
+  from libcloudforensics.providers.azure.internal import compute  # pylint: disable=cyclic-import
+
+# pylint: disable=line-too-long
+# See https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
+# pylint: enable=line-too-long
+REGEX_DISK_NAME = re.compile('^[\\w]{1,80}$')
+REGEX_SNAPSHOT_NAME = re.compile('^(?=.{1,80}$)[a-zA-Z0-9]([\\w,-]*[\\w])?$')
+
+DEFAULT_DISK_COPY_PREFIX = 'evidence'
+
+LOGGER = logging.getLogger()
+
+
+def GetCredentials() -> ServicePrincipalCredentials:
+  """Get Azure credentials.
+
+  Returns:
+    ServicePrincipalCredentials: Azure credentials.
+  """
+  return ServicePrincipalCredentials(tenant=os.environ["AZURE_TENANT_ID"],
+                                     client_id=os.environ["AZURE_CLIENT_ID"],
+                                     secret=os.environ["AZURE_CLIENT_SECRET"])
+
+
+def ExecuteRequest(
+    client: Any,
+    func: str,
+    kwargs: Optional[Dict[str, str]] = None) -> List[Any]:
```

Comment:
Why not use `List[Dict]` here if that's what you're returning?

Reply:
Ah. remnants of copy-paste. I updated the docstring.